homeassistant/components/ecobee/climate.py

```diff
@@ -726,6 +740,77 @@ def resume_program(self, resume_all):
         )
         self.update_without_throttle = True
 
+    def set_active_sensors(
+        self, sensors: list[str], climate_name: str | None = None
+    ) -> None:
+        """Set the currently active sensors on a thermostat."""
+        if climate_name is None:
+            climate_name = self.preset_mode
+
+        # Get device name from device id.
+        device_registry = dr.async_get(self.hass)
+        sensor_names: list[str] = []
+        for sensor in sensors:
+            sensor_name = device_registry.async_get(sensor).name
+            sensor_names.append(sensor_name)
+
+        # Check if climate is an available preset option.
+        if climate_name not in self._preset_modes.values():
+            _LOGGER.error(
+                f"Invalid climate name, available options are: "
+                f"{', '.join(self.preset_modes)}"
+            )
+            raise HomeAssistantError(
+                f"Invalid climate name, available options are: "
+                f"{', '.join(self.preset_modes)}"
+            )
```

Comment:
```suggestion
            msg = f"Invalid climate name, available options are: {', '.join(self.preset_modes)}"
            _LOGGER.error(msg)
            raise HomeAssistantError(msg)
```

Please move this step before you access the device registry

Reply:
Changed to msg format (sorry about that... should have noticed it was dumb to do it that way). Moved this check before the device registry lookup.